docs/ADR-X-cardano-api-exports-convention.md

```diff
@@ -0,0 +1,43 @@
+# Status
+
+📜 Proposed
+
+# Context
+
+# Proposed Design
+
+## Top level exports
+
+Each top level export should export distinct symbols.
+This may depend on the case, for example: `Cardano.Api.Byron` symbols which are still in use in current eras, can be reexported through `Cardano.Api`.
+
+* `Cardano.Api` - everything domain related for the currently used and upcoming eras
+* `Cardano.Api.E` - everything for the obsoleted `E` era, for example `Cardano.Api.Byron`.
```

Comment:
If you are saying we should have `Cardano.Api.Allegra`, `Cardano.Api.Mary` ... etc then I disagree. We do have to keep `Cardano.Api.Byron` however.

We should have a `Cardano.Api.Compatible` top level module that is dictated by the needs of our QA team and perhaps the community if need be.

Reply:
Yes, good point. I think we should stick to `Compatible` + `Byron`. Separate eras modules won't be very useful.